solidity/contracts/bridge/Bridge.sol

```diff
@@ -264,34 +323,309 @@ contract Bridge {
+        }
+
+        uint256 observedDiff = bitcoinHeaders.validateHeaderChain();
+
+        require(
+            observedDiff != ValidateSPV.getErrBadLength(),
+            "Invalid length of the headers chain"
+        );
+        require(
+            observedDiff != ValidateSPV.getErrInvalidChain(),
+            "Invalid headers chain"
+        );
+        require(
+            observedDiff != ValidateSPV.getErrLowWork(),
+            "Insufficient work in a header"
+        );
+
+        require(
+            observedDiff >= requestedDiff * txProofDifficultyFactor,
+            "Insufficient accumulated difficulty in header chain"
+        );
+    }
+
+    // TODO: Documentation.
+    function processSweepTxOutput(BitcoinTx.Info calldata sweepTx)
+        internal
+        pure
+        returns (bytes20 walletPubKeyHash, uint64 value)
+    {
+        // To determine the total number of sweep transaction outputs, we need to
+        // parse the compactSize uint (VarInt) the output vector is prepended by.
+        // That compactSize uint encodes the number of vector elements using the
+        // format presented in:
+        // https://developer.bitcoin.org/reference/transactions.html#compactsize-unsigned-integers
+        // We don't need asserting the compactSize uint is parseable since it
+        // was already checked during `validateVout` validation.
+        (, uint256 outputsCount) = sweepTx.outputVector.parseVarInt();
+        require(
+            outputsCount == 1,
+            "Sweep transaction must have a single output"
+        );
+
+        bytes memory output = sweepTx.outputVector.extractOutputAtIndex(0);
+        value = output.extractValue();
+        bytes memory walletPubKeyHashBytes = output.extractHash();
+        // The sweep transaction output should always be P2PKH or P2WPKH.
+        // In both cases, the wallet public key hash should be 20 bytes length.
+        require(
+            walletPubKeyHashBytes.length == 20,
+            "Wallet public key hash should have 20 bytes"
+        );
+        /* solhint-disable-next-line no-inline-assembly */
+        assembly {
+            walletPubKeyHash := mload(add(walletPubKeyHashBytes, 32))
+        }
+
+        return (walletPubKeyHash, value);
+    }
+
+    // TODO: Documentation.
+    function processSweepTxInputs(
+        BitcoinTx.Info calldata sweepTx,
+        bytes20 walletPubKeyHash
+    )
+        internal
+        returns (
+            uint256 inputsValue,
+            address[] memory depositors,
+            uint256[] memory depositedAmounts
+        )
+    {
+        SweepInfo storage previousSweep = sweeps[walletPubKeyHash];
+        // Flag indicating whether the previous sweep transaction output
+        // is included as an input. If current sweep is the first sweep
+        // of given wallet, we use a shortcut and initialize it with `true`.
+        bool previousSweepFlag = previousSweep.txHash == bytes32(0);
```

Comment:
Could we give it a more clear name, so that we know that if this flag is set to true there was no previous sweeps?
Perhaps `isFirstSweep` or `noPreviousSweeps`.

Reply:
I had a hard time trying to get a proper name here. This flag doesn't only indicate whether we should expect the previous sweep but also indicates whether the previous sweep reference was found in case it is expected (there is a prior sweep for a given wallet). Normally, we should have two separate variables but it's Solidity and I stumbled upon `stack too deep` errors which forced me to reduce the number of local variables. This is why I gave it such a generic name which should be interpreted as: "is the previous sweep condition fulfilled?" which means either the previous sweep is not required OR it is required and found.